cfme/tests/infrastructure/test_delete_infra_object.py

```diff
@@ -109,14 +109,13 @@ def test_delete_datastore(setup_provider, provider_crud, remove_test):
     test_datastore = datastore.Datastore(name=data_store)
     host_count = test_datastore.get_hosts()
     vm_count = test_datastore.get_vms()
-    if len(host_count) == 0 and len(vm_count) == 0:
-        test_datastore.delete(cancel=False)
-    else:
+    if len(host_count) != 0:
```

Comment:
sequences with length > 0 evaluate as True; This and line 115 should be `if host_count:`